pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -185,10 +205,25 @@ def on_pretrain_routine_start(self, trainer, pl_module):
         self.__resolve_ckpt_dir(trainer)
         self.save_function = trainer.save_checkpoint
 
+    def on_train_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx) -> None:
+        if self._should_skip_saving_checkpoint(trainer):
+            return
+        step = trainer.global_step
```

Comment:
Should be `trainer.total_batch_idx` as `global_step` only counts optimizer_step and they would be different if accumulated_grad_batches is activated.